### PR TITLE
Adding mfpymake package built with greyskull

### PR DESCRIPTION
**Pull request by jdhughes-usgs**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/mfpymake/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  number: 0
+
+requirements:
+  host:
+    - python >=3.7
+    - setuptools >=45
+    - wheel
+    - pip
+  run:
+    - python >=3.8
```
Comment (dopplershift) [inline recipes/mfpymake/meta.yaml:28]: Any reason these two should be different?

File: recipes/mfpymake/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - pymake
+  commands:
+    - mfpymake --help
+    - make-program --help
+    - make-code-json --help
+
+about:
+  home: https://github.com/modflowpy/pymake
+  summary: pymake is a Python package to compile MODFLOW-based models.
+  license: Unlicense
```
Comment (dopplershift) [inline recipes/mfpymake/meta.yaml:47]: Looks like it actually should be:
```suggestion
  license: CC0-1.0
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just a couple minor tweaks and this is good to go. I also need @w-bonelli to confirm that they are willing to be listed as a maintainer for this recipe.

File: recipes/mfpymake/meta.yaml (hunk 1 above)
Comment (jdhughes-usgs) [reply recipes/mfpymake/meta.yaml]: No. Both are python >= 3.8 now

**Comment by w-bonelli** [comment]
> Just a couple minor tweaks and this is good to go. I also need @w-bonelli to confirm that they are willing to be listed as a maintainer for this recipe.

That is fine with me. Thanks

**Review by dopplershift — APPROVED** [review]

**Pull request merged**